src/shared/sam/localLambdaRunner.ts

```diff
@@ -202,52 +209,118 @@ export async function invokeLambdaFunction(
 
     await onAfterBuild()
 
-    ctx.chanLogger.info(
-        'AWS.output.starting.sam.app.locally',
-        'Starting the SAM Application locally (see Terminal for output)'
-    )
+    ctx.chanLogger.info('AWS.output.starting.sam.app.locally', 'Starting SAM application locally')
     getLogger().debug(`localLambdaRunner.invokeLambdaFunction: ${config.name}`)
 
     const maxRetries: number = getAttachDebuggerMaxRetryLimit(ctx.settings, MAX_DEBUGGER_RETRIES_DEFAULT)
+    const timer = createInvokeTimer(ctx.settings)
+    const debugPort = !config.noDebug ? config.debugPort?.toString() : undefined
+
+    if (config.invokeTarget.target === 'api') {
+        // sam local start-api ...
+        const samCliContext = new DefaultSamCliProcessInvokerContext()
+        const sam = await samCliContext.cliConfig.getOrDetectSamCli()
+        if (!sam.path) {
+            getLogger().warn('SAM CLI not found and not configured')
+        } else if (sam.autoDetected) {
+            getLogger().info('SAM CLI not configured, using SAM found at: %O', sam.path)
+        }
+        const samCommand = sam.path ? sam.path : 'sam'
+        const samArgs = await buildSamCliStartApiArguments({
+            templatePath: config.templatePath,
+            dockerNetwork: config.sam?.dockerNetwork,
+            environmentVariablePath: config.envFile,
+            environmentVariables: envVars,
+            port: config.apiPort?.toString(),
+            debugPort: debugPort,
+            debuggerPath: config.debuggerPath,
+            debugArgs: config.debugArgs,
+            skipPullImage: config.sam?.skipNewImageCheck,
+            parameterOverrides: config.parameterOverrides,
+            extraArgs: config.sam?.localArguments,
+        })
 
-    const localInvokeArgs: SamCliLocalInvokeInvocationArguments = {
-        templateResourceName: makeResourceName(config),
-        templatePath: config.templatePath,
-        eventPath: config.eventPayloadFile,
-        environmentVariablePath: config.envFile,
-        environmentVariables: envVars,
-        invoker: config.samLocalInvokeCommand!,
-        dockerNetwork: config.sam?.dockerNetwork,
-        debugPort: !config.noDebug ? config.debugPort?.toString() : undefined,
-        debuggerPath: config.debuggerPath,
-        debugArgs: config.debugArgs,
-        extraArgs: config.sam?.localArguments,
-        parameterOverrides: config.parameterOverrides,
-        skipPullImage: config.sam?.skipNewImageCheck,
-    }
-
-    const command = new SamCliLocalInvokeInvocation(localInvokeArgs)
+        function recordApigwTelemetry(result: telemetry.Result) {
```

Comment:
Is there a reason why we have this function embedded in another function? Can it be pulled out (we typically only put functions as consts inside other functions if we're passing some sort of callback functionality)? If not, our style has been `const recordApigwTelemetry = (result: telemetry.Result) => {`

Reply:
It has not purpose outside of this function. It is a local variable. The fact that it's a function doesn't mean it belongs in module scope.

Our existing habits in this regard should be changed.

Reply:
That's fine, in that case swap it to `const x = <arrow function>` terminology. It's recommended to use `function () {}` as a method, and arrow functions as non-method functions.